src/source/PeerConnection/Rtcp.c

```diff
@@ -139,6 +139,138 @@ static STATUS onRtcpReceiverReport(PRtcpPacket pRtcpPacket, PKvsPeerConnection p
+{
+    /*
+        0                   1                   2                   3
+        0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |V=2|P|  FMT=15 |    PT=205     |           length              |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |                     SSRC of packet sender                     |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |                      SSRC of media source                     |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |      base sequence number     |      packet status count      |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |                 reference time                | fb pkt. count |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |          packet chunk         |         packet chunk          |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       .                                                               .
+       .                                                               .
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |         packet chunk          |  recv delta   |  recv delta   |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       .                                                               .
+       .                                                               .
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+       |           recv delta          |  recv delta   | zero padding  |
+       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+     */
+    STATUS retStatus = STATUS_SUCCESS;
+    UINT32 packetsRemaining;
+    UINT16 baseSeqNum, packetStatusCount, packetSeqNum;
+    UINT32 chunkOffset, recvOffset;
+    UINT8 statusSymbol;
+    UINT32 packetChunk;
+    INT16 recvDelta;
+    UINT32 statuses;
+    UINT32 i;
+    CHK(pKvsPeerConnection != NULL && pRtcpPacket != NULL, STATUS_NULL_ARG);
+    // dont parse if neither callbacks are set
+    CHK(pKvsPeerConnection->onPacketNotReceived != NULL || pKvsPeerConnection->onPacketReceived != NULL, STATUS_SUCCESS);
+
+    baseSeqNum = getUnalignedInt16BigEndian(pRtcpPacket->payload + 8);
+    packetStatusCount = TWCC_PACKET_STATUS_COUNT(pRtcpPacket->payload);
+
+    packetsRemaining = packetStatusCount;
+    chunkOffset = 16;
+    while (packetsRemaining > 0 && chunkOffset < pRtcpPacket->payloadLength) {
+        packetChunk = getUnalignedInt16BigEndian(pRtcpPacket->payload + chunkOffset);
+        if (IS_TWCC_RUNLEN(packetChunk)) {
+            packetsRemaining -= TWCC_RUNLEN_GET(packetChunk);
+        } else {
+            packetsRemaining -= MIN(TWCC_STATUSVECTOR_COUNT(packetChunk), packetsRemaining);
```

Comment:
Is it not possible that the value becomes less than 0 and the 2s complement gets stored leading to a very large value and hence a potential infinite loop? To avoid the issue, I would suggest to change the loop this way:

```
while (packetsRemaining < packetStatusCount && chunkOffset < pRtcpPacket->payloadLength) {
        packetChunk = getUnalignedInt16BigEndian(pRtcpPacket->payload + chunkOffset);
        if (IS_TWCC_RUNLEN(packetChunk)) {
            packetsRemaining += TWCC_RUNLEN_GET(packetChunk);
        } else {
            packetsRemaining += MIN(TWCC_STATUSVECTOR_COUNT(packetChunk), packetsRemaining);
        }
        chunkOffset += 2;
    }
```

Modify the naming appropriately.

Reply:
good catch

Reply:
i made packetsRemaining signed int, the `while (packetsRemaining > 0` should take care of  the infinite loop